app/javascript/shared/components/useKeyboardShortcuts.jsx

```diff
@@ -31,19 +36,68 @@ function isFormField(element) {
  *   "ctrl+alt+KeyG": (e) => {
  *     e.preventDefault();
  *     alert("Control Alt G has been pressed");
+ *   },
+  *   "KeyG~KeyH": (e) => {
+ *     e.preventDefault();
+ *     alert("G has been pressed quickly followed by H");
  *   },
  *   "?": (e) => {
  *     setIsHelpVisible(true);
  *   }
  * }
  *
- * useKeyboardShortcuts(shortcuts, someElementOrWindowObject);
+ * useKeyboardShortcuts(shortcuts, someElementOrWindowObject, {timeout: 1500});
  *
  * @param {object} shortcuts List of keyboard shortcuts/event
  * @param {EventTarget} [eventTarget=window] An event target.
+ * @param {object} [options = {}] An object for extra options
  *
  */
-export function useKeyboardShortcuts(shortcuts, eventTarget = window) {
+export function useKeyboardShortcuts(shortcuts, eventTarget = window, options = {}) {
+  const [keyChain, setKeyChain] = useState([]);
+  const [keyChainQueue, setKeyChainQueue] = useState(null);
```

Comment:
refactor (question): Just wondering if `keyChainQueue` is necessary. Why not just append to `keyChain` via `setKeyChain`?

Reply:
I was avoiding having `keyChain` in the useEffect that set and unset the eventListener to prevent it being added and removed each key press.